source/scale/ensuring-releases-perform-at-scale.rst

```diff
@@ -0,0 +1,45 @@
+Ensuring Releases Perform at Scale
+==================================
+
+To ensure each release of Mattermost upholds our high standards for performance at scale, the Mattermost Engineering team performs thorough load testing, develops features with scale in mind, and follows strict guidelines for database schema migrations.
+
+Monthly release load tests
+--------------------------
+
+Each month, before being approved for distribution, a release candidate of Mattermost is load tested via a comprehensive and mature set of `load test tooling <https://github.com/mattermost/mattermost-load-test-ng>`__ using simulated data that matches real-world, high-scale usage patterns.
+
+Multiple tests covering different configurations are run against highly-available deployments of Mattermost with thousands of users and millions of posts. Both PostgreSQL and MySQL are tested, although MySQL tests will be dropped when :ref:`MySQL goes out of support with Mattermost v11 <about/deprecated-features:mattermost server v10.0.0>`.
+
+The load tests generate a report detailing average API request times, database I/O, memory usage, concurrency, requests per second, and more. This performance report on the release candidate is then compared to the previous month's report. Any deviations are investigated and remedied before the release candidate is promoted to the final release.
```

Comment:
> This performance report on the release candidate is then compared to the previous month's report.

This is slightly misleading: we don't compare against previous reports, but against the latest version. That is: we run a test and generate a report on the latest stable version, and the same test and generate another report on the latest RC version to be released; e.g., `v9.11.0` vs `v10.0.0-rc2` (https://github.com/mattermost/performance-reports/tree/main/performance-comparisons/v10.0). Then we compare those two reports. The analysis that we publish (like the one I just linked is then on the comparison between those two reports.

Reply:
Thanks! That's good clarification. How's the update look?

Reply:
That's perfect! Thanks :)